frontend/kesaseteli/employer/src/components/header/Header.tsx

```diff
@@ -0,0 +1,84 @@
+import useLogin from 'kesaseteli/employer/hooks/useLogin';
+import useLogoutQuery from 'kesaseteli/employer/hooks/useLogoutQuery';
+import useUserQuery from 'kesaseteli/employer/hooks/useUserQuery';
+import { useRouter } from 'next/router';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+import BaseHeader from 'shared/components/header/Header';
+import useLocale from 'shared/hooks/useLocale';
+import { SUPPORTED_LANGUAGES } from 'shared/i18n/i18n';
+import { OptionType } from 'shared/types/common';
+
+const Header: React.FC = () => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const router = useRouter();
+  const { asPath } = router;
+  const getLanguageOptions = (): OptionType[] => {
+    const createOptions = (languages: readonly string[]): OptionType[] =>
+      languages.map((language) => ({
+        label: t(`common:languages.${language}`),
+        value: language,
+      }));
+    return createOptions(SUPPORTED_LANGUAGES);
+  };
+
+  const languageOptions: OptionType[] = getLanguageOptions();
+
+  const handleLanguageChange = (
+    e: React.SyntheticEvent<unknown>,
+    { value: lang }: OptionType
+  ): void => {
+    e.preventDefault();
+    void router.push(asPath, asPath, {
```

Comment:
eslint enforces it. It tells the compiler that there is purposely a floating promise that we dont want to `await` for. Eslint rule is generally a good one, because at least I have several times encoutered a hard-to-find bug when forgot to `await` the promise.

Reply:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md

Reply:
👍